### PR TITLE
feat: LANZ-2647 Add namespaces and IAM

### PR DESCRIPTION
**Pull request by jstehlik-ohpen**

### CODE REVIEW

File: providers.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = "~> 1.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 3.0"
```
Comment (mliner) [inline providers.tf:7]: version = "~> 3.0"

**Review by mliner — CHANGES_REQUESTED** [review]

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,55 @@
+variable "namespace_type" {
+  type = string
+  description = "(Required) Type of namespace"
+
+  validation {
+    condition     = contains(["http", "private_dns", "public_dns"], var.namespace_type)
```
Comment (lucky-edu) [inline variables.tf:6]: I think is safer to go first with ONLY http. There's already a lot of testing to do aroung that. Easier to review also :).

**Review by lucky-edu — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 2 above)
Comment (jstehlik-ohpen) [reply variables.tf:6]: @lucky-edu testing results for all scenarios can be found here
https://ohpen.atlassian.net/browse/LANZ-2647
spoiler alert - it works

File: data.tf [hunk 3]
```diff
@@ -0,0 +1 @@
+data "aws_caller_identity" "current" {}
```
[the remark this answers is not in the document]
Comment (jstehlik-ohpen) [reply data.tf:1]: @lucky-edu
Used only in default value if role is supposed to be created, but no allowed accounts are specified
https://github.com/ohpensource/terraform-aws-ohp-cloudmap-namespace/blob/b4eb446049af7a41e44226fff95d3d0bc982fdc0/locals.tf#L34

**Review by lucky-edu — APPROVED** [review]

**Pull request merged**